SCClassLibrary/Common/Collections/linux/extString_linux.sc

```diff
@@ -1,6 +1,26 @@
 + String {
-	runInTerminal {|shell="sh"|
-		("xterm -hold -T 'SuperCollider runInTerminal' -e" + shell + "-c" + this.shellQuote).unixCmd;
+
+	runInTerminal { |shell="/bin/sh"|
+		LinuxPlatform.runInTerminalCmd ?? { LinuxPlatform.runInTerminalCmd = LinuxPlatform.findTerminalEmulator };
+		if(LinuxPlatform.runInTerminalCmd.isNil) {
+			"String:runInTerminal couldn't find a supported terminal emulator. Please set LinuxPlatform.runInTerminalCmd, see Help for details".error;
+			^ this;
```

Comment:
this is dead code; `error` throws so `^this` will never be reached. also, for future reference we usually don't put a space after `^`

Reply:
oh! you're right. in that case, i think this branch should actually throw, what do you think?

Reply:
>  i think this branch should actually throw, what do you think?

Theoretically yes, but on macOS we don't throw when there's a problem with temporary file creation, we similarly post the error without throwing.
While I don't have strong opinion, I'd indeed lean towards throwing an error and, in a separate PR, updating macOS implementation to also throw if there's a problem.

Reply:
After reading the next comment, I left this as it was, calling .error and not throwing (like on mac). I liked the idea of throwing though, but I wouldn't like the same code to throw on one platform and not on another, so I think an eventual subsequent PR should change both implementations at the same time :)